### PR TITLE
MCR-2436 added tests for fact based acl system

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2436).

### CODE REVIEW

File: mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCategoryConditionTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        MCRCategoryCondition categoryCondition = new MCRCategoryCondition();
+
+        categoryCondition.parse(new Element("classification").setText("clazz:clazz2"));
+        Assert.assertFalse("Object not should be linked with clazz2",
+                categoryCondition.matches(holder));
+
+    }
+
+}
```
Comment (rsteph-de) [inline mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCategoryConditionTest.java]: Github already recognized "No linebreak" at the end
Please "auto-format" all new files

File: mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCreatedByConditionTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        object.getService().addFlag("createdby", "administrator");
+        MCRObjectID testId = MCRObjectID.getInstance("test_test_00000001");
+
+        MCRFactsHolder holder = new MCRFactsHolder(new ArrayList<>());
+        hackObjectIntoCache(object, testId);
+        holder.add(new MCRObjectIDFact("objid", testId.toString(), testId));
+
+        MCRCreatedByCondition createdByCondition = new MCRCreatedByCondition();
+        createdByCondition.parse(new Element("createdBy"));
+
```
Comment (rsteph-de) [inline mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCreatedByConditionTest.java:60]: I changed the element name to "createdby" some minutes ago

File: mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRIPConditionTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Test
+    public void testConditionNotMatch() {
+        MCRSessionMgr.getCurrentSession().setCurrentIP("222.222.222.222");
+        MCRIPCondition condition = new MCRIPCondition();
+        condition.parse(new Element("ip").setText("111.111.111.111"));
+        MCRFactsHolder facts = new MCRFactsHolder(Collections.emptyList());
+        Assert.assertFalse("The ip should not match", condition.matches(facts));
+    }
```
Comment (rsteph-de) [inline mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRIPConditionTest.java:57]: please add some tests for IP ranges
for example: 111.111.0.0./255.255.0.0

File: mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRStateConditionTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        object.getService().setState("published");
+        MCRObjectID testId = MCRObjectID.getInstance("test_test_00000001");
+
+        MCRFactsHolder holder = new MCRFactsHolder(new ArrayList<>());
+        hackObjectIntoCache(object, testId);
+        holder.add(new MCRObjectIDFact("objid", testId.toString(), testId));
+
+        MCRStateCondition mcrStateCondition = new MCRStateCondition();
+        mcrStateCondition.parse(new Element("state").setText("published"));
+        Assert.assertTrue("State should be published!", mcrStateCondition.matches(holder));
```
Comment (rsteph-de) [inline mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRStateConditionTest.java]: change to "State should be 'published'!"
also in following method

File: mycore-mods/src/test/java/org/mycore/mods/access/facts/condition/MCRMODSGenreConditionTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        mw.setMODS(new Element("mods", MCRConstants.MODS_NAMESPACE));
+        MCRObjectID testId = MCRObjectID.getInstance("test_mods_00000001");
+        object.setId(testId);
+
+        MCRFactsHolder holder = new MCRFactsHolder(new ArrayList<>());
+        hackObjectIntoCache(object, testId);
+        holder.add(new MCRObjectIDFact("objid", testId.toString(), testId));
+
+        Element classification = mw.addElement("genre");
+        classification.setAttribute("valueURI", "https://mycore.de/classifications/mir_genres#article");
```
Comment (rsteph-de) [inline mycore-mods/src/test/java/org/mycore/mods/access/facts/condition/MCRMODSGenreConditionTest.java]: rename 'classification' to 'genre'

**Review by rsteph-de — CHANGES_REQUESTED** [review]
some minor changes required

see comments!

File: mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCreatedByConditionTest.java (hunk 2 above)
Comment (sebhofmann) [reply mycore-base/src/test/java/org/mycore/access/facts/condition/fact/MCRCreatedByConditionTest.java:60]: the element names dont matter for the tests.

**Review by rsteph-de — APPROVED** [review]
looks good for me now!

**Pull request merged**